### PR TITLE
Static linking and backtrace after Swift 5.9

### PR DESCRIPTION
**Pull request by vzsg**
With Swift 5.9 release, crash reporting basically disappeared overnight on Heroku, as the previous backtrace library became no-op, and the new system needs extra steps during build.

This changeset adds those missing steps, but by extension breaks compatibility with older Swift versions. But that should be okay. Anything except the very latest minor release and snapshots version is dead in the water anyway.

If upgrading Swift to 5.10.1 or 6.0 is non-negotiable, I recommend sticking with the old version of the buildpack with:

```
heroku buildpacks:set https://github.com/vapor-community/heroku-buildpack.git#5101-release
```

Or the equivalent setting on the Heroku dashboard.

### CODE REVIEW

**Comment by vzsg** [comment]
Actually, I'll revive the swiftly branch instead.
Juggling swiftenv is just too annoying.

File: bin/compile [hunk 1]
```diff
@@ -65,3 +65,4 @@ mkdir -p $BUILD_DIR/.profile.d
 
 set-env PATH '$HOME/.swift-bin:$PATH'
 set-env LD_LIBRARY_PATH '$LD_LIBRARY_PATH:$HOME/.swift-lib'
+set-env SWIFT_BACKTRACE 'enable=yes,sanitize=yes,threads=all,images=all,interactive=no,output-to=stderr,symbolicate=full,swift-backtrace=.swift-bin/swift-backtrace-static'
```
Comment (MahdiBM) [inline bin/compile:68]: I think `symbolicate=fast` is better. It usually finishes in only a few seconds (most i've seen is 5).
`full` can take minutes, like 2-3+. Then there is the chance that heruko just kills the node and users end up with no backtrace which will pretty confusing.

cc @gwynne i'd vote for this change in the template repo too.
Perhaps with a comment to mention `full` exists, above the env.
Or even better, just link to docs right there.

File: bin/steps/swift-build [hunk 2]
```diff
@@ -1,2 +1,2 @@
 puts-step "Building package ($SWIFT_BUILD_CONFIGURATION configuration)"
-swift build $SWIFT_BUILD_FLAGS -c "$SWIFT_BUILD_CONFIGURATION"
+swift build $SWIFT_BUILD_FLAGS -c "$SWIFT_BUILD_CONFIGURATION" --static-swift-stdlib
```
Comment (MahdiBM) [inline bin/steps/swift-build:2]: We should also use jemalloc but that should be another PR.
It shouldn't be as problematic.

Comment (vzsg) [reply bin/steps/swift-build:2]: Good idea, actually.

**Comment by vzsg** [comment]
As swiftly does not play along with Ubuntu 24.04, I'll keep this one after all...

File: bin/compile (hunk 1 above)
Comment (vzsg) [reply bin/compile]: I switched it to `fast`.

In my limited testing there was no significant difference, but then I only tested it with a bare template with extra fatalErrors, so I'll trust you on this one.

File: bin/steps/swift-build (hunk 2 above)
Comment (vzsg) [reply bin/steps/swift-build:2]: That integration will more likely come in the form of a section in the README, as installing jemalloc cleanly is a solved problem: https://github.com/brian-kephart/heroku-buildpack-jemalloc.
